src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,16 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public Boolean isLeapYear(int year) {
+        if (year >= 1 && year <= 9999) {
+            if (year % 4 == 0 && year % 100 != 0) {
```

Comment:
No need to separate condition in if(year % 4 == 0 && year % 100 != 0) and else if(year % 100 == 0 && year % 400 == 0)
they both return true, so what is the idea?

you can simplify the logic and readability
```
// check incorrect input
        if (year < 1 || year > 9999) {
            return false;
        }
        // check all the leap year conditions here
        if () {
            return true;
        }
        return false;
```